src/theory/quantifiers/sygus/rcons_obligation_info.h

```diff
@@ -0,0 +1,131 @@
+/*********************                                                        */
+/*! \file rcons_obligation_info.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Abdalrhman Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief utility class for Sygus Reconstruct module
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__QUANTIFIERS__RCONS_OBLIGATION_INFO_H
+#define CVC4__THEORY__QUANTIFIERS__RCONS_OBLIGATION_INFO_H
+
+#include "expr/node.h"
+
+namespace CVC4 {
+namespace theory {
+namespace quantifiers {
+
+/**
+ * A utility class for Sygus Reconstruct obligations. An obligation is a
+ * builtin term t and a SyGuS type T, and indicates that we are trying to build
+ * a term of type T whose builtin analog is equivalent to t.
+ */
+class RConsObligationInfo
+{
+ public:
+  /**
+   * Constructor. Default value needed for maps.
+   *
+   * @param builtin builtin term to reconstruct
+   */
+  explicit RConsObligationInfo(Node builtin = Node::null());
```

Comment:
To avoid implicit conversion...

Reply:
It'd be good to add to the documentation why it's used here, since this is not common in the rest of the code.

Reply:
I believe it should be the other way around. People should explain why they're **_not_** using the explicit keyword. Implicit conversions are generally frowned upon because they reduce code documentation and may introduce hard to detect bugs.

Reply:
Fair enough. :) Regarding the general issue of using it or not, we can discuss this at some of the CVC4 meetings. We could update our guidelines accordingly.